core/dbt/parser/models.py

```diff
@@ -26,46 +27,106 @@ def render_update(
     ) -> None:
         self.manifest._parsing_info.static_analysis_path_count += 1
 
-        # normal dbt run
-        if not flags.USE_EXPERIMENTAL_PARSER:
-            super().render_update(node, config)
+        # True roughly 1/100 times this function is called
+        sample: bool = random.randint(1, 101) == 100
 
-        # if the --use-experimental-parser flag was set
-        else:
+        # run the experimental parser if the flag is on or if we're sampling
+        if flags.USE_EXPERIMENTAL_PARSER or sample:
             try:
-                # run dbt jinja extractor (powered by tree-sitter + rust)
-                # throws an exception if it can't parse the source
-                res = py_extract_from_source(node.raw_sql)
+                experimentally_parsed = py_extract_from_source(node.raw_sql)
 
-                # since it doesn't need python jinja, fit the refs, sources, and configs
-                # into the node. Down the line the rest of the node will be updated with
-                # this information. (e.g. depends_on etc.)
+                # second config format
                 config_calls = []
-                for c in res['configs']:
+                for c in experimentally_parsed['configs']:
```

Comment:
Hmm I'm seeing some misalignments in the sense that the experimental parser returns
```
[{'unique_key': 'event_id'}, {'materialized': 'incremental'}]
```
Whereas traditional parsing returns
```
[{'unique_key': 'event_id', 'materialized': 'incremental'}]
```
With the resulting value of: `['02_false_positive_config_value', '03_missed_config_value']`

Reply:
yup. I think the above comment should address this now. Thanks for the detailed example!

Reply:
I was wrong, this needed to be handled completely separately. Thanks for pointing it out!